[triton] Unify build_paths.py for NV & AMD, fix typing

danzimm opened the pull request:
Summary: Some build improvements.

Test Plan: CI

Differential Revision: D63583959


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

danzimm commented:
@pytorchbot label "topic: not user facing"

bertmaher commented on torch/_inductor/cpp_builder.py:
```diff
@@ -705,24 +703,21 @@ def _setup_standard_sys_libs(
-        include_dirs.append(build_paths.libgcc_backward())
-        include_dirs.append(build_paths.glibc())
-        include_dirs.append(build_paths.linux_kernel())
+            # TODO(T203136598): Is there any harm in including sleef_include in the hip path?
```
Yeah this is weird, I can't imagine why sleef causes a problem for hip

danzimm committed:
[triton] Unify build_paths.py for NV & AMD, fix typing (#136952)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/136952

Some build improvements.

Test Plan: CI

Reviewed By: bertmaher, nmacchioni

Differential Revision: D63583959

izaitsevfb commented:
@pytorchbot merge -f 'Lint issue is not related'
